doc/aaa/TACACS+ Requirement.md

```diff
@@ -0,0 +1,161 @@
+| exec     | Specifying a filename containing a slash as an argument to the -p option. |
+| deleting/adding     | Use the -f and -d options to the enable builtin. |
+| enable     | Using the 'enable' builtin command to enable disabled shell builtins. |
+| command     | Specifying the -p option to the 'command' builtin command. |
+| set     | Turning off restricted mode with ‘set +r’ or ‘set +o restricted’. |
+
+	- All these behavior disabled only for user input, command in script will not be affected, see here for more details: https://www.gnu.org/software/bash/manual/html_node/The-Restricted-Shell.html
+
+
+- Supported Authorization types:
+	- EXEC: user session authorization support. this happen when user login.
+	- Command: user run command in shell.
+
+- Failover:
+	- Authorization will happen before execute, if remote TACACS server not available, use local group based authorization as failover.
+
+
+## 1.3 Accounting
+ - Accounting when:
+ 	- User login to SONiC host.
+ 	- User run command on host: 
+		- Command start run.
+		- Command finish.
+		- For recursive command/script, only the top level command have Accounting.
+- Failover:
+	- Use syslog as backup when remote TACACS not avaliable.
+
+
+## 1.4 User script
+ - Any script in whitelist can run with Authorization and Accounting. 
+ - If user create a script, admin user can use config command add script to whitelist.
+
+## 1.5 Docker support
+ - Docker exec command will be covered by Authorization and Accounting. 
+ - SONiC AAA can't cover any command user run inside a docker.
+
+## 1.6 Multiple TACACS server
+ - Support config multiple TACACS server.
+ - When a server not available, will try next server as backup.
+ - When all server not available, use native failover solution.
+
+# 2 Configuration and Management Requirements
+## 2.1 SONiC CLI
+ - Enable/Disable TACACS Authorization/Accounting command
+```
+    config tacacs authorization enable
+    config tacacs authorization disable
+    config tacacs accounting enable
+    config tacacs accounting disable
+```
+
+ - Whitelist management command
+```
+    show config tacacs restrict command whitelist
+    Config tacacs restrict command whitelist remove <command>
+    Config tacacs restrict command whitelist add <command>
+```
+
+ - Counter command
+```
+    show tacacs counter
+    clear tacacs counter
+```
+
+## 2.2 Config DB
+ - TACACS AAA are fully configable by config DB.
+
+## 2.3 Counter
+ - Support  AAA counter:
+```
+    show tacacs counter
+    
+        server1: 10.1.1.45
+        Messages sent: 24
+        Messages received: 20
+        Requests accepted: 14
+        Requests rejected: 8
+        Requests timeout: 2
+        Requests retransmitted: 1
+        Bad responses: 1
+```
+
+## 2.4 System log
+- Generate system log when Authentication/Authorization/Accounting.
+- When remote TACACS server not avaliable, use system log for accounting.
```

Comment:
>When remote TACACS server not avaliable [](http://example.com/codeflow?start=2&length=39)

When tacacs server is accessible, do we also use local syslog to account?

Reply:
I think it's not necessary.
If tacacs server is accessible, we only put trace log to syslog for metering and debugging.

Reply:
I guess your `system log` == `syslog`.
Then if the networking is not stable, you will have some random log lines locally, does it help anything?

For login authentication, the log messages always go to local syslog, no matter remote tacacs accessible or not.

Reply:
Document updated.

What I mean here is when the remote TACACS service not accessible, for example TACACS server down after user login, but networking still healthy, then the Authorization and Accounting information will write to syslog. Then the syslog will upload via network later, and those information will not lost.